plugin/evm/vm_test.go

```diff
@@ -770,44 +753,357 @@ func TestIssueAtomicTxs(t *testing.T) {
+	require.NoError(err)
+
+	err = vm.issueTx(importTx, true)
+	require.NoError(err)
+
+	<-issuer
+
+	blk, err := vm.BuildBlock(context.Background())
+	require.NoError(err)
+
+	atomicTxBlock := blk.(*chain.BlockWrapper).Block.(*Block)
+
+	err = blk.Verify(context.Background())
+	require.NoError(err)
+
+	require.Equal(blk.Status(), choices.Processing, "Expected status to match")
+
+	err = vm.SetPreference(context.Background(), blk.ID())
+	require.NoError(err)
+
+	err = blk.Accept(context.Background())
+	require.NoError(err)
+
+	require.Equal(blk.Status(), choices.Accepted, "Expected status to match")
+
+	lastAcceptedID, err := vm.LastAccepted(context.Background())
+	require.NoError(err)
+	require.Equal(lastAcceptedID, blk.ID(), "Expected last accepted blockID to be the accepted block")
+
+	newHead := <-newTxPoolHeadChan
+	require.Equal(newHead.Head.Hash(), common.Hash(blk.ID()), "Expected new block to match")
+
+	// Create a block that has no atomic txs
+	tx := types.NewTransaction(0, testEthAddrs[0], big.NewInt(0), 21000, big.NewInt(params.LaunchMinGasPrice), nil)
+	signedTx, err := types.SignTx(tx, types.NewEIP155Signer(vm.chainID), testKeys[0].ToECDSA())
+	require.NoError(err)
+
+	errs := vm.txPool.AddRemotesSync([]*types.Transaction{signedTx})
+	for _, err := range errs {
+		require.NoError(err)
+	}
+
+	<-issuer
+
+	blk2, err := vm.BuildBlock(context.Background())
+	require.NoError(err)
+
+	noAtomicTxBlock := blk2.(*chain.BlockWrapper).Block.(*Block)
+
+	// All new blocks will be post apricot5
+	vm.chainConfig.ApricotPhase5BlockTimestamp = utils.NewUint64(0)
+
+	// Create a post apricot5 block that has 1 atomic tx
+	importTx2, err := vm.newImportTx(vm.ctx.XChainID, testEthAddrs[1], initialBaseFee, []*secp256k1.PrivateKey{testKeys[1]})
+	require.NoError(err)
+
+	err = vm.issueTx(importTx2, true)
+	require.NoError(err)
+
+	<-issuer
+
+	blk3, err := vm.BuildBlock(context.Background())
+	require.NoError(err)
+
+	atomicTxBlockPostApricot5 := blk3.(*chain.BlockWrapper).Block.(*Block)
+
+	// Create a post apricot5 block that has no atomic txs
+	tx = types.NewTx(&types.DynamicFeeTx{
+		Nonce:     1,
+		To:        &testEthAddrs[0],
+		Value:     common.Big0,
+		Gas:       50000,
+		GasFeeCap: new(big.Int).Mul(big.NewInt(100), big.NewInt(params.GWei)),
+		GasTipCap: big.NewInt(10),
+		Data:      nil,
+	})
+	signedTx, err = types.SignTx(tx, types.NewLondonSigner(vm.chainID), testKeys[0].ToECDSA())
+	require.NoError(err)
+
+	errs = vm.txPool.AddRemotesSync([]*types.Transaction{signedTx})
+	for _, err := range errs {
+		require.NoError(err)
+	}
+
+	<-issuer
+
+	vm.clock.Set(vm.clock.Time().Add(2 * time.Second))
+	blk4, err := vm.BuildBlock(context.Background())
+	require.NoError(err)
+
+	noAtomicTxBlockPostApricot5 := blk4.(*chain.BlockWrapper).Block.(*Block)
+
+	// Create a post apricot5 block that has multiple atomic txs
+	kc := secp256k1fx.NewKeychain()
+	kc.Add(testKeys[0])
+	txID, err := ids.ToID(hashing.ComputeHash256(testShortIDAddrs[0][:]))
```

Comment:
not sure I follow why this is called txID?

Reply:
copied from another test, changing name

Reply:
I believe this is called txID to match the input name string for `addUtxo` . I'll keep this name.

Reply:
Looks like this is copied from addUTXO at the start of a test where it could actually just be a randomly generated UTXOID instead. Can we switch to using `ids.GenerateTestID()` here and fix it above as well instead of copying my old mistake 😅